src/GitVersionCore/GitVersionCache.cs

```diff
@@ -43,56 +43,70 @@ public void WriteVariablesToDiskCache(IRepository repo, string gitDir, VersionVa
-                            vv = VersionVariables.FromFile(cacheFileName, fileSystem);
+                            fileSystem.Delete(cacheFileName);
                         }
-                        catch (Exception ex)
+                        catch (Exception deleteEx)
                         {
-                            Logger.WriteWarning("Unable to read cache file " + cacheFileName + ", deleting it.");
-                            Logger.WriteInfo(ex.ToString());
-                            try
-                            {
-                                fileSystem.Delete(cacheFileName);
-                            }
-                            catch (Exception deleteEx)
-                            {
-                                Logger.WriteWarning(string.Format("Unable to delete corrupted version cache file {0}. Got {1} exception.", cacheFileName, deleteEx.GetType().FullName));
-                            }
+                            Logger.WriteWarning(string.Format("Unable to delete corrupted version cache file {0}. Got {1} exception.", cacheFileName, deleteEx.GetType().FullName));
                         }
+
+                        return null;
                     }
                 }
-                else
-                {
-                    Logger.WriteInfo("Cache file " + cacheFileName + " not found.");
-                }
-
-                return vv;
             }
         }
 
-        string GetKey(IRepository repo, string gitDir)
+        string GetKey(GitPreparer gitPreparer)
         {
+            var dotGitDirectory = gitPreparer.GetDotGitDirectory();
+
             // Maybe using timestamp in .git/refs directory is enough?
-            var ticks = fileSystem.GetLastDirectoryWrite(Path.Combine(gitDir, "refs"));
-            var configPath = Path.Combine(repo.GetRepositoryDirectory(), "GitVersionConfig.yaml");
-            var configText = fileSystem.Exists(configPath) ? fileSystem.ReadAllText(configPath) : null;
-            var configHash = configText != null ? GetHash(configText) : null;
-            return string.Join(":", gitDir, repo.Head.CanonicalName, repo.Head.Tip.Sha, ticks, configHash);
+            var lastGitRefsChangedTicks = fileSystem.GetLastDirectoryWrite(Path.Combine(dotGitDirectory, "refs"));
+
+            // will return the same hash even when config file will be moved 
+            // from workingDirectory to rootProjectDirectory. It's OK. Config essentially is the same.
+            var configFilePath = ConfigurationProvider.SelectConfigFilePath(gitPreparer, fileSystem);
+            var configFileContent = fileSystem.Exists(configFilePath) ? fileSystem.ReadAllText(configFilePath) : null;
+            var configFileHash = configFileContent != null ? GetHash(configFileContent) : null;
+
+            return gitPreparer.WithRepository(repo => string.Join(":", dotGitDirectory, repo.Head.CanonicalName, repo.Head.Tip.Sha, lastGitRefsChangedTicks, configFileHash));
```

Comment:
It doesn't required! We will ignore default override config from arguments when it was not actually overriden


SpecifiedArgumentRunner.Run
```
 var overrideConfig = arguments.HasOverrideConfig ? arguments.OverrideConfig : null;

            var executeCore = new ExecuteCore(fileSystem);
            var variables = executeCore.ExecuteGitVersion(targetUrl, dynamicRepositoryLocation, authentication, targetBranch, noFetch, targetPath, commitId, overrideConfig: overrideConfig);
```

**When it was overriden we shouldn't persist in cache results of it. Or should?**

ExecuteCore.ExecuteGitVersion
```
if (overrideConfig != null)
            {
                using (Logger.IndentLog("Override config from command line"))
                {
                    var overridenVersionVariables = ExecuteInternal(targetBranch, commitId, gitPreparer, buildServer, overrideConfig: overrideConfig);
                    return overridenVersionVariables;
                }
            }

            var versionVariables = gitVersionCache.LoadVersionVariablesFromDiskCache(gitPreparer);
```